Scripts/Python/ki/xKIChat.py

```diff
@@ -1016,10 +1016,18 @@ def GetPID(self, params):
             return int(params)
         except ValueError:
             for player in self.chatMgr.BKPlayerList:
+                # Age Players
                 if isinstance(player, ptPlayer):
                     plyrName = player.getPlayerName().casefold()
                     if params.casefold() == plyrName:
                         return player.getPlayerID()
+                # Buddies or Neighbors
+                elif isinstance(player, ptVaultNodeRef):
+                    plyrInfoNode = player.getChild().upcastToPlayerInfoNode()
+                    if type(plyrInfoNode) != type(None) and plyrInfoNode.getType() == PtVaultNodeTypes.kPlayerInfoNode:
```

Comment:
```suggestion
                    if plryInfoNode is not None:
```

`upcastToPlayerInfoNode()` will only ever return `None` objects or `ptVaultPlayerInfoNode` objects whose node type is `kPlayerInfoNode` by definition.